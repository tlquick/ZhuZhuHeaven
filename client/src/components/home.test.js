import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "./home";
afterEach(() => {
  cleanup();
});
test("renders home page", () => {
  render(<Home />);
  const homeElement = screen.getByTestId("home-1");
  expect(homeElement).toBeInTheDocument();
  expect(homeElement).toContainHTML("div"); //contains a <div> tag
});
test("matches home snapshot", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
