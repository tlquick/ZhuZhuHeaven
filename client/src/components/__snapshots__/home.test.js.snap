// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home snapshot 1`] = `
<div>
  <div
    className="main-screen mx-auto"
    data-testid="home-1"
  >
    <div
      className="main-screen container-fluid mx-2"
      data-testid="category-1"
    >
      <div
        className="row mt-5 mx-2"
      >
        <div
          className="col-md-3"
        >
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            All
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 1
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 2
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 3
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 4
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 5
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 6
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Series 7
          </button>
          <button
            className="btn btn-info w-100 mb-4"
            onClick={[Function]}
          >
            Special Edition
          </button>
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="row"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
